src/test/java/com/diffmin/AppTest.java

```diff
@@ -197,9 +209,46 @@ void should_apply_pure_move_patches(TestResources sources) throws Exception {
         runTests(sources);
     }
 
+    private static boolean doesElementBelongToSpecifiedFile(
+            CtElement element, String filePathPrefix) {
+        return element.getPosition().getFile().getName().startsWith(filePathPrefix);
+    }
+
+    private static boolean doesElementBelongToModifiedSet(CtElement element) {
+        if (element.getRoleInParent() == CtRole.THROWN) {
+            return ((CtExecutable<?>) element.getParent())
+                    .getThrownTypes().stream()
+                            .anyMatch(
+                                    thrownType ->
+                                            doesElementBelongToSpecifiedFile(
+                                                    thrownType, NEW_PREFIX));
+        }
+        return false;
+    }
+
+    private static boolean isChildOfInsertedPath(
+            String elementPathString, Set<String> newRevisionPathStrings) {
+        return newRevisionPathStrings.stream()
+                .anyMatch(
+                        modifiedPathString ->
+                                !(modifiedPathString.equals(elementPathString))
+                                        && elementPathString.startsWith(modifiedPathString));
+    }
+
+    private static boolean skipAssertionCheck(
+            CtElement element, Set<String> newRevisionPathStrings) {
+        if (element == null || element.isImplicit() || !element.getPosition().isValidPosition()) {
+            return true;
+        }
```

Comment:
This should of course also go into the `ElementSourceFileChecker` class (or whatever you call it).